packages/core/src/State.ts

```diff
@@ -290,9 +290,12 @@ export class State<
    */
   public matches<TSV extends TTypestate['value']>(
     parentStateValue: TSV
-  ): this is TTypestate extends { value: TSV }
-    ? State<TTypestate['context'], TEvent, TStateSchema, TTypestate>
-    : never {
+  ): this is State<
+    (TTypestate extends { value: TSV } ? TTypestate : never)['context'],
```

Comment:
this introduces a regression on `.context` property after `.matches` call:
```ts
type TContext = {foo: number}

const toggleMachine = createMachine<TContext>({
  id: 'toggle',
  initial: 'a',
  states: {
    a: { on: { TOGGLE: 'b' } },
    b: { on: { TOGGLE: 'a' } }
  }
});

const state = toggleMachine.initialState;

if (state.matches('a')) {
  // context is now `any` but was correctly preserved as TContext before this change
  const { context } = state
}
```

Reply:
Resolved!